iree/compiler/Dialect/HAL/Target/LLVM/LLVMAOTTarget.cpp

```diff
@@ -253,8 +253,10 @@ class LLVMAOTTargetBackend final : public TargetBackend {
       int64_t localMemorySize = entryPointOp.workgroup_local_memory()
                                     .getValueOr(APInt(64, 0))
                                     .getSExtValue();
+      // Grab source IR code here to pass it down as a string for dispatch user tags.
+      const char* source_code = "test tag";
```

Comment:
By this point, within `serializeExecutable()`, higher level ops have already been processed into backend-specific code, so any summarization should probably happen earlier. One idea was to add a new pass earlier in compilation that would generate the tag for each executable and store it in an attribute, then pull the value out of that attribute down here.

That could happen as early as after `createOutlineDispatchRegionsPass` in the `flow` dialect transformation pass pipeline:

https://github.com/google/iree/blob/e5b269a0b5229d459682a4593f8d992b2f60eb2a/iree/compiler/Dialect/Flow/Transforms/Passes.cpp#L144-L181

and as late as before `createTranslateExecutablesPass` in the `hal` dialect transformation pass pipeline:
https://github.com/google/iree/blob/e5b269a0b5229d459682a4593f8d992b2f60eb2a/iree/compiler/Dialect/HAL/Transforms/Passes.cpp#L195-L241

You can look through the IR at each stage by running with `-print-ir-after-all` and/or `-print-ir-before-all`. That output can get large though. Here's a random sample I ran recently on a small program: https://gist.github.com/ScottTodd/5e85347201a90ecf2328aa278f987f9b.

* `flow.executable` first appears after `OutlineDispatchRegions`. More complex programs could have linalg named ops in there, but this one just has a `linalg.generic` around an elementwise op
* `ConvertToHALPass` (after translation) has a `hal.executable.variant` filled with `llvm` dialect ops that are too low level for good summarization

I have some other useful flags and IR dumps for another effort collected here if that helps: https://gist.github.com/ScottTodd/3cae2022f98242163ba44dfe6fd73fad

Reply:
This helped me a lot! Just working through build errors now.